pkg/pool-linear/test/AaveLinearPoolFactory.test.ts

```diff
@@ -202,4 +219,41 @@ describe('AaveLinearPoolFactory', function () {
       expect(bufferPeriodEndTime).to.equal(now);
     });
   });
+
+  describe('protocol id', () => {
+    context('with no registered protocols', () => {
+      it('should revert when asking for an unregistered protocol name', async () => {
+        await expect(factory.getProtocolName(AAVE_PROTOCOL_ID)).to.be.revertedWith('Protocol not registered');
+      });
+
+      it('should not allow adding protocols without permission', async () => {
+        await expect(factory.registerProtocolId(AAVE_PROTOCOL_ID, 'AAVE')).to.be.revertedWith('SENDER_NOT_ALLOWED');
+      });
+    });
+
+    context('with registered protocols', () => {
+      sharedBeforeEach('grant permissions', async () => {
+        const action = await actionId(factory, 'registerProtocolId');
+        await vault.authorizer.connect(admin).grantPermissions([action], admin.address, [factory.address]);
+      });
+
+      sharedBeforeEach('register some protocols', async () => {
+        await factory.connect(admin).registerProtocolId(AAVE_PROTOCOL_ID, AAVE_PROTOCOL_NAME);
+        await factory.connect(admin).registerProtocolId(BEEFY_PROTOCOL_ID, BEEFY_PROTOCOL_NAME);
+        await factory.connect(admin).registerProtocolId(STURDY_PROTOCOL_ID, STURDY_PROTOCOL_NAME);
+      });
+
+      it('should register protocols', async () => {
+        expect(await factory.getProtocolName(AAVE_PROTOCOL_ID)).to.equal(AAVE_PROTOCOL_NAME);
+        expect(await factory.getProtocolName(BEEFY_PROTOCOL_ID)).to.equal(BEEFY_PROTOCOL_NAME);
+        expect(await factory.getProtocolName(STURDY_PROTOCOL_ID)).to.equal(STURDY_PROTOCOL_NAME);
+      });
+
+      it('should fail when a protocol is already registered', async () => {
+        await expect(
+          factory.connect(admin).registerProtocolId(STURDY_PROTOCOL_ID, 'Rando protocol')
```

Comment:
```suggestion
          factory.connect(admin).registerProtocolId(STURDY_PROTOCOL_ID, 'Random protocol')
```

Reply:
I actually did intend "rando" - it means random, but also has the connotation of being suspicious / potentially dangerous. Maybe that's too much slang :) I can see how people might think it's a misspelling of random.